rust/chains/hyperlane-sealevel/src/interchain_gas.rs

```diff
@@ -49,12 +79,144 @@ impl InterchainGasPaymaster for SealevelInterchainGasPaymaster {}
 
 /// Struct that retrieves event data for a Sealevel IGP contract
 #[derive(Debug)]
-pub struct SealevelInterchainGasPaymasterIndexer {}
+pub struct SealevelInterchainGasPaymasterIndexer {
+    rpc_client: RpcClientWithDebug,
+    igp: SealevelInterchainGasPaymaster,
+    program_id: Pubkey,
+    sealevel_config: Option<SealevelConf>,
+}
+
+#[derive(Debug, new)]
+pub struct SealevelGasPayment {
+    payment: InterchainGasPayment,
+    log_meta: LogMeta,
+    beneficiary: H256,
+}
 
 impl SealevelInterchainGasPaymasterIndexer {
     /// Create a new Sealevel IGP indexer.
-    pub fn new(_conf: &ConnectionConf, _locator: ContractLocator) -> Self {
-        Self {}
+    pub fn new(
+        conf: &ConnectionConf,
+        locator: ContractLocator,
+        sealevel_config: Option<SealevelConf>,
+    ) -> Self {
+        let program_id = Pubkey::from(<[u8; 32]>::from(locator.address));
+        // Set the `processed` commitment at rpc level
+        let rpc_client = RpcClientWithDebug::new_with_commitment(
+            conf.url.to_string(),
+            CommitmentConfig::processed(),
+        );
+        let igp = SealevelInterchainGasPaymaster::new(conf, locator);
+        Self {
+            program_id,
+            rpc_client,
+            igp,
+            sealevel_config,
+        }
+    }
+
+    async fn get_payment_with_sequence(
+        &self,
+        sequence_number: u64,
+    ) -> ChainResult<SealevelGasPayment> {
+        let payment_bytes = &[
+            &hyperlane_sealevel_igp::accounts::GAS_PAYMENT_DISCRIMINATOR[..],
```

Comment:
Why the slice operators?

Reply:
Because `Concat` is implemented for slices but not for arrays, and there is no auto-coercion that can make this work AFAICT